### DISCUSSION TITLE
Installation

### DISCUSSION THREAD

**Discussion by voterpublic** [RondeSC/Miller_Shuffle_Algo#3] [Q&A] [answered]
Hi there. I am a real novice so bare with me.
I thank you for creating this. 
I am very interested because I often listen to music on a random playback option however the algorithms used are prone to repeating the exact same songs again and again. I use a raspberry pi with either MoodeAudio (http://moodeaudio.org/) or rAudio-1 (https://github.com/rern/rAudio-1) installed. Both have random options however neither is good at random play. Fixes have been implemented by both but when switching to random play the very same songs are always played time after time.

Can I install your randomiser into an installation of either MoodeAudio (or rRune or other you know of) on raspberry pi hardware?
If so, how (please)?
Many thanks

**Comment by RondeSC** [comment]
I have no experience with any of the software you speak of. I have been using MSAlgo in connection with all sorts of games and the giving out training materials.
In a month or so I may investigate this, if no one else integrates a MSA shuffle into one of them.

**Comment by RondeSC** [comment, marked as answer]
If upon starting, "the very same songs are always played", and not just some occurrences of repeats, it's likely not the shuffle algorithm but the random (PRNG) function. All shuffle algorithms are to some degree dependent on a system random function. As a PRNG function is not random at all, it must be initialized with a random seed value based on an independent event. 
When you start the player or switch to ‘shuffle’ you could reseed the PRNG (e.g. using ‘srand()’) with the value of the system millisecond clock (not automatically from 'cron').

**Comment by voterpublic** [comment]
So cool Ron. Thank you for answering my query and taking the time to think
about it.
I will try what you suggested. It is worth a go.
I may have incorrectly stated 'exact same songs' a little. It is more the
very same songs in a differnent order from the same artists/album. Some
artists/albums never play at all. It the played songs always come from the
same album.
Many thanks again for your time and consideration
Lou

no logo
